### PR TITLE
Added Mocha, Chai and ESLint

### PR DESCRIPTION
**Pull request by angelkbrown**
Added 
- Test framework [Mocha](https://mochajs.org/)
- Assertion library [Chai](https://www.chaijs.com/)
- Linter [ESLint](https://eslint.org/)
- Also added some basic tests for the existing routes and added test command to `scripts` definition of `package.json`
  - Run `npm run test` to run the test suite

### CODE REVIEW

**Comment by angelkbrown** [comment]
To match the frontend, we may want to add the commit linter tools, Prettier and Typescript, which Darga added to the frontend:
> [Husky](https://typicode.github.io/husky/)
> [CommitLint](https://commitlint.js.org/#/)
> [Commitizen](https://commitizen-tools.github.io/commitizen/)
> [Prettier](https://prettier.io/)
> [Typescript](https://www.typescriptlang.org/)

File: test/routes.js
Comment (dargaCode) [inline test/routes.js:1]: Nice, I've never seen how to do tests for a server before!

File: package.json [hunk 1]
```diff
@@ -16,6 +17,15 @@
...
   "devDependencies": {
+    "chai": "^4.3.10",
+    "chai-http": "^4.4.0",
+    "eslint": "^8.52.0",
+    "eslint-config-semistandard": "^17.0.0",
+    "eslint-config-standard": "^17.1.0",
+    "eslint-plugin-import": "^2.29.0",
+    "eslint-plugin-n": "^15.7.0",
+    "eslint-plugin-promise": "^6.1.1",
+    "mocha": "^10.2.0",
```
Comment (dargaCode) [inline package.json:28]: Thank you for adding these! There's one more step to get ESlint reporting on the updated rules, which is to `extend` these configs in the `.eslintrc`

I opened a PR into your branch with these changes: 

https://github.com/Code-the-Dream-School/ee-prac-team2-back/pull/2

**Review by dargaCode — APPROVED** [review]
LGTM! We can either treat that other PR as part of this branch, or merge this and then I'll retarget the other PR to `main`

**Pull request merged**